core/designsystem/src/main/java/com/teamhy2/designsystem/ui/theme/Theme.kt

```diff
@@ -0,0 +1,88 @@
+package com.teamhy2.designsystem.ui.theme
+
+import android.app.Activity
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.darkColorScheme
+import androidx.compose.material3.lightColorScheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.SideEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.platform.LocalView
+import androidx.core.view.WindowCompat
+
+private val DarkColorScheme = darkColorScheme(
+	primary = Blue100,
+	secondary = Blue400,
+	tertiary = Yellow100,
+	background = Black
+)
+
+private val LightColorScheme = lightColorScheme(
+	primary = Blue100,
+	secondary = Blue400,
+	tertiary = Yellow100,
+	background = Black
+)
+
+private val LocalHY2Typography = staticCompositionLocalOf<HY2Typography> {
+	error("No HY2Typography provided")
+}
+
+/*
+* HY2Theme
+*
+* Typo를 변경하고 싶다면 HY2Theme.typography.title01 방식으로 접근
+* */
+object HY2Theme {
+	val typography: HY2Typography
+		@Composable
+		@ReadOnlyComposable
+		get() = LocalHY2Typography.current
+}
+
+@Composable
+fun ProvideHY2Typography(
+	typography: HY2Typography,
+	content: @Composable () -> Unit
+) {
+	val provideTypography = remember { typography.copy() }.apply { update(typography) }
+	CompositionLocalProvider(
+		LocalHY2Typography provides provideTypography,
+		content = content
+	)
+}
+
+@Composable
+fun HY2Theme(
+	darkTheme: Boolean = isSystemInDarkTheme(),
+	content: @Composable () -> Unit
+) {
+	val colorScheme = if (darkTheme) DarkColorScheme else LightColorScheme
+	val view = LocalView.current
+	if (!view.isInEditMode) {
+		SideEffect {
+			val window = (view.context as Activity).window
+			val insetsController = WindowCompat.getInsetsController(window, view)
+			val statusBarColor = colorScheme.background.toArgb()
+			window.statusBarColor = statusBarColor
+
+			// 항상 다크테마이기 때문에 상태바 아이콘들을 밝은 톤으로 표시
+			insetsController.isAppearanceLightStatusBars = false
+		}
+	}
+
+	val typography = HY2Typography()
+
+	ProvideHY2Typography(typography) {
+		MaterialTheme(
+			colorScheme = colorScheme,
+			content = content
+		)
+	}
```

Comment:
`typography` 지역 변수가 사용하는 쪽에 인라인 되어도 괜찮을거 같은데 혹시 다른 이유가 있을까요?

Reply:
HY2Colors의 추후 추가 가능성을 생각해 보았을 때 가독성 측면에서 좋지 않을까 생각이 들었습니다!

```suggestion
	val typography = HY2Typography()
	val colors = HY2Colors()

	ProvideHY2Typography(typography, colors) {
		MaterialTheme(
			content = content
		)
	}
```

Reply:
가독성에 있어서는 취향차이로 보이지만 성능적으로는 근소하게나마 지역변수를 사용하는 쪽이 불리하다고 생각이 듭니다!
`ProvideHY2Typography(HY2Typography(), HY2Colors())` 
이렇게 쓰는 것도 그다지 나빠보이진 않는데 어떤가요?

Reply:
네 좋습니다 반영하겠습니다!